conanfile.py

```diff
@@ -17,38 +18,75 @@ class WinflexbisonConan(ConanFile):
     exports = ["LICENSE.md"]
     exports_sources = ["CMakeLists.txt"]
     generators = "cmake"
-    settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False]}
-    default_options = {"shared": False}
-    _source_subfolder = "source_subfolder"
+    settings = "arch", "os_build", "arch_build", "compiler", "build_type"
 
-    def config_options(self):
-        if self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("winflexbison is only supported on Windows.")
+    _source_subfolder = "sources"
+
+    def configure(self):
+        self.settings.arch = str(self.settings.arch_build)
+        if self.settings.compiler != "Visual Studio":
+            raise ConanInvalidConfiguration("winflexbison is only supported for Visual Studio.")
+
+    def package_id(self):
+        self.info.include_build_settings()
+        del self.info.settings.arch
+        del self.info.settings.compiler
+        del self.info.settings.build_type
 
     def source(self):
-        tools.get("{0}/archive/v{1}.tar.gz".format(self.homepage, self.version), sha256="39fe57de6a52dc83c8a9ece90b8484d8d2b764e24e22e30ba5dc018328019a4d")
-        extracted_dir = self.name + "-" + self.version
+        name = "winflexbison"
+        url = "https://github.com/lexxmark/winflexbison/archive/v{}.tar.gz".format(self.version)
+        sha256 = "2ab4c895f9baf03dfdfbb2dc4abe60e87bf46efe12ed1218c38fd7761f0f58fc"
+        filename = "{}-{}.tar.gz".format(name, self.version)
+
+        dlfilepath = os.path.join(tempfile.gettempdir(), filename)
+        if os.path.exists(dlfilepath) and not tools.get_env("WINFLEXBISON_FORCE_DOWNLOAD", False):
+            self.output.info("Skipping download. Using cached {}".format(dlfilepath))
+        else:
+            self.output.info("Downloading {} from {}".format(name, url))
+            tools.download(url, dlfilepath)
+        tools.check_sha256(dlfilepath, sha256)
+        tools.untargz(dlfilepath)
+
+        extracted_dir = "{}-{}".format(name, self.version)
```

Comment:
This is way too much custom logic, please stay with the templates.

If you need to make modifications there should be a good reason.
If these are general reasons please make pull requests to the templates

Reply:
Ok, no more caching.